docs/guides/batch_norm.rst

```diff
@@ -0,0 +1,294 @@
+Using BatchNorm
+===============
+
+In this guide we will go through the details of using ``BatchNorm`` in models,
+in the process we will highlight some of the differences between code that uses
+``BatchNorm`` and code that does not.
+
+.. testsetup::
+
+  import flax.linen as nn
+  import jax.numpy as jnp
+  import jax
+  import optax
+  from typing import Any
+  from flax.core import FrozenDict
+
+Defining the model
+******************
+
+``BatchNorm`` is a Module that has different runtime behavior between training and
+inference. In other frameworks this behavior specified via mutable state, however
```

Comment:
Could you maybe add a link pointing to another framework and how it uses mutable state? It is a bit unclear to me now.

Reply:
Added 2 links, and some additional comments.